### ISSUE TITLE
sgemm stauts code -2048` for `test_dnn_conv.simple_gpu_conv_backward_filters` on hd530

### ISSUE THREAD

**Issue by hughperkins** [hughperkins/cuda-on-cl#17]
```
[ RUN      ] test_dnn_conv.simple_gpu_conv_backward_filters
outH=5 outW=6
forward workspaceSizeBytes=3240
backward filter workspaceSizeBytes=5400
workspaceSizeBytes=5400
CL_GPUOFFSET var detected, changing gpu offset from 0 to 0
Using Apple , OpenCL platform: Apple
Using OpenCL device: Intel(R) HD Graphics 530
gpuMemoryAllocSize=10340
building kernel im2col_kernel
 ... built
sgemm status code -2048
unknown file: Failure
Unknown C++ exception thrown in the test body.
```

**Comment by iamqk** [comment]
same failed on iMac (2.7 GHz Intel Core i5, Intel Iris Pro ), os x 10.12.5, on git 2f2f3b5

build warning include kKernelLaunchError as flowed, hope can offer some help

`[ 62%] Building CXX object CMakeFiles/clblast.dir/src/CLBlast/src/routines/level3/xsyrk.cpp.o
/Volumes/Data/work/DeepLearning/tensorflow/tensorflow-cl/third_party/cuda-on-cl/src/CLBlast/src/routines/level3/xsymm.cpp:111:16: warning: 
      27 enumeration values not handled in switch: 'kKernelLaunchError', 'kKernelRunError',
      'kInvalidLocalMemUsage'... [-Wswitch]
        switch(status) {
               ^
/Volumes/Data/work/DeepLearning/tensorflow/tensorflow-cl/third_party/cuda-on-cl/src/CLBlast/src/routines/level3/xsymm.cpp:111:16: warning: 
      27 enumeration values not handled in switch: 'kKernelLaunchError', 'kKernelRunError',
      'kInvalidLocalMemUsage'... [-Wswitch]
/Volumes/Data/work/DeepLearning/tensorflow/tensorflow-cl/third_party/cuda-on-cl/src/CLBlast/src/routines/level3/xsymm.cpp:111:16: warning: 
      27 enumeration values not handled in switch: 'kKernelLaunchError', 'kKernelRunError',
      'kInvalidLocalMemUsage'... [-Wswitch]
/Volumes/Data/work/DeepLearning/tensorflow/tensorflow-cl/third_party/cuda-on-cl/src/CLBlast/src/routines/level3/xsymm.cpp:111:16: warning: 
      27 enumeration values not handled in switch: 'kKernelLaunchError', 'kKernelRunError',
      'kInvalidLocalMemUsage'... [-Wswitch]
/Volumes/Data/work/DeepLearning/tensorflow/tensorflow-cl/third_party/cuda-on-cl/src/CLBlast/src/routines/level3/xsymm.cpp:111:16: warning: 
      27 enumeration values not handled in switch: 'kKernelLaunchError', 'kKernelRunError',
      'kInvalidLocalMemUsage'... [-Wswitch]
/Volumes/Data/work/DeepLearning/tensorflow/tensorflow-cl/third_party/cuda-on-cl/src/CLBlast/src/routines/level3/xhemm.cpp:111:16: warning: 
      27 enumeration values not handled in switch: 'kKernelLaunchError', 'kKernelRunError',
      'kInvalidLocalMemUsage'... [-Wswitch]
        switch(status) {
               ^
/Volumes/Data/work/DeepLearning/tensorflow/tensorflow-cl/third_party/cuda-on-cl/src/CLBlast/src/routines/level3/xhemm.cpp:111:16: warning: 
      27 enumeration values not handled in switch: 'kKernelLaunchError', 'kKernelRunError',
      'kInvalidLocalMemUsage'... [-Wswitch]
2 warnings generated.
[ 63%] Building CXX object CMakeFiles/clblast.dir/src/CLBlast/src/routines/level3/xherk.cpp.o
5 warnings generated.
[ 64%] Building CXX object CMakeFiles/clblast.dir/src/CLBlast/src/routines/level3/xsyr2k.cpp.o
[ 65%] Building CXX object CMakeFiles/clblast.dir/src/CLBlast/src/routines/level3/xher2k.cpp.o
[ 66%] Building CXX object CMakeFiles/clblast.dir/src/CLBlast/src/routines/level3/xtrmm.cpp.o
[ 67%] Building CXX object CMakeFiles/clblast.dir/src/CLBlast/src/routines/levelx/xomatcopy.cpp.o
/Volumes/Data/work/DeepLearning/tensorflow/tensorflow-cl/third_party/cuda-on-cl/src/CLBlast/src/routines/level3/xtrmm.cpp:114:16: warning: 
      27 enumeration values not handled in switch: 'kKernelLaunchError', 'kKernelRunError',
      'kInvalidLocalMemUsage'... [-Wswitch]
        switch(status) {
               ^
/Volumes/Data/work/DeepLearning/tensorflow/tensorflow-cl/third_party/cuda-on-cl/src/CLBlast/src/routines/level3/xtrmm.cpp:114:16: warning: 
      27 enumeration values not handled in switch: 'kKernelLaunchError', 'kKernelRunError',
      'kInvalidLocalMemUsage'... [-Wswitch]
/Volumes/Data/work/DeepLearning/tensorflow/tensorflow-cl/third_party/cuda-on-cl/src/CLBlast/src/routines/level3/xtrmm.cpp:114:16: warning: 
      27 enumeration values not handled in switch: 'kKernelLaunchError', 'kKernelRunError',
      'kInvalidLocalMemUsage'... [-Wswitch]
/Volumes/Data/work/DeepLearning/tensorflow/tensorflow-cl/third_party/cuda-on-cl/src/CLBlast/src/routines/level3/xtrmm.cpp:114:16: warning: 
      27 enumeration values not handled in switch: 'kKernelLaunchError', 'kKernelRunError',
      'kInvalidLocalMemUsage'... [-Wswitch]
/Volumes/Data/work/DeepLearning/tensorflow/tensorflow-cl/third_party/cuda-on-cl/src/CLBlast/src/routines/level3/xtrmm.cpp:114:16: warning: 
      27 enumeration values not handled in switch: 'kKernelLaunchError', 'kKernelRunError',
      'kInvalidLocalMemUsage'... [-Wswitch]
5 warnings generated.
[ 68%] Linking CXX shared library libclblast.dylib
`

**Comment by iamqk** [comment]
test log

`check gpu image vs col2im:
image val 0.696469 after col2im 2.78588
unknown file: Failure
Unknown C++ exception thrown in the test body.
[  FAILED  ] test_dnn_conv.simple_gpu_im2col (1409 ms)`

**Comment by hughperkins** [comment]
Yes. Need to prod Cedric really, ie https://github.com/CNugteren/CLBlast/issues/149 .  If you have a moment to work through the various tests/diagnostics he recommends, that could be very useful :-)

**Comment by hughperkins** [comment]
(Note: you can ignore the build warnings.  They're just warnings, to me, but they're ok.)

**Comment by hughperkins** [comment]
Could do with someone to go through the various diagnostics for this issue with Cedric, if someone has a moment?
